data-serving/reusable-data-service/data_service/model/case.py

```diff
@@ -94,13 +103,18 @@ def validate(self):
 # so Case is the class that you should use.
 
 
-def make_custom_case_class(name: str, fields=[]) -> type:
-    """Generate a class extending the DayZeroCase class with additional fields."""
+def make_custom_case_class(name: str, fields=[], field_models=[]) -> type:
+    """Generate a class extending the DayZeroCase class with additional fields.
+    fields is a list of dataclass fields that should be added to the generated class.
+    field_models is a list of model objects describing the fields for the data dictionary
+    and for validation."""
+    # FIXME generate the fields list from the field_models
```

Comment:
my plan for this is when the day zero schema is committed, to create a list of `Field` objects that contain the day zero fields and get played into the `Case` class; then the logic for `make_custom_case_class` is simply to take a list of all known `Field`s and make data class fields for them. At that point I think the `DayZeroCase` intermediate class goes away.